### PR TITLE
feat: saved query list actions

### PR DESCRIPTION
**Pull request by riahk**
### SUMMARY
- [x] Add query edit, query link copy, query delete action functionality
- [x] Add bulk select/delete functionality
- [x] Create `ActionsBar` component

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2020-09-29 at 3 59 06 PM
Screen Shot 2020-09-29 at 3 59 12 PM
Screen Shot 2020-09-29 at 3 59 28 PM

### TEST PLAN
- [x] Update `SavedQueryList_spec.js`

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/TabbedSqlEditors.jsx [hunk 1]
```diff
@@ -109,6 +109,8 @@ class TabbedSqlEditors extends React.PureComponent {
       ...URI(window.location).search(true),
     };
 
+    console.log('query', query);
```
Comment (nytai) [inline superset-frontend/src/SqlLab/components/TabbedSqlEditors.jsx:112]: do we still want to keep this log?

File: superset-frontend/spec/javascripts/views/CRUD/data/savedquery/SavedQueryList_spec.jsx [hunk 2]
```diff
@@ -108,6 +117,51 @@ describe('SavedQueryList', () => {
     );
   });
 
+  it('renders ActionsBar in table', () => {
```
Comment (nytai) [inline superset-frontend/spec/javascripts/views/CRUD/data/savedquery/SavedQueryList_spec.jsx:120]: Thanks for adding tests!

**Review by nytai — COMMENTED** [review]
mostly LGTM, just a couple questions/nits

File: superset-frontend/src/views/CRUD/data/savedquery/SavedQueryList.tsx [hunk 3]
```diff
@@ -299,17 +450,92 @@ function SavedQueryList({
...
+              data={queries}
+              fetchData={fetchData}
+              filters={filters}
+              initialSort={initialSort}
+              loading={loading}
+              pageSize={PAGE_SIZE}
+              bulkActions={bulkActions}
+              bulkSelectEnabled={bulkSelectEnabled}
+              disableBulkSelect={toggleBulkSelect}
+              renderBulkSelectCopy={selected => {
```
[the remark this answers is not in the document]
Comment (nytai) [reply superset-frontend/src/views/CRUD/data/savedquery/SavedQueryList.tsx:499]: The ListView component already generates a default copy, this prop just just for overriding it (needed for datasets)

**Review by willbarrett — COMMENTED** [review]
One nit, otherwise looking good!

File: superset-frontend/src/components/ListView/ActionsBar.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+/* type ActionProps = {
+  label: string;
+  tooltip?: string | React.ReactElement;
+  placement?: string;
+  icon: IconName | string;
+  onClick: () => void;
+}; */
+
+type ActionsBarProps = {
+  actions: Array<any>;
```
[the remark this answers is not in the document]
Comment (riahk) [reply superset-frontend/src/components/ListView/ActionsBar.tsx:33]: I was having some issues with typescript when I tried that, mainly with the specific `IconName` type we use. I'll try again, might have missed something.

[the remark this answers is not in the document]
Comment (nytai) [reply superset-frontend/src/components/ListView/ActionsBar.tsx:33]: Can you try importing the `IconProps` and referencing `IconProps["name"]`?

**Review by nytai — APPROVED** [review]

**Pull request merged**